algorithms/src/std_algorithms/impl/Kokkos_ForEachForEachN.hpp

```diff
@@ -41,29 +65,45 @@ struct StdForEachFunctor {
       : m_first(std::move(_first)), m_functor(std::move(_functor)) {}
 };
 
-template <class ExecutionSpace, class IteratorType, class UnaryFunctorType>
-UnaryFunctorType for_each_impl(const std::string& label,
-                               const ExecutionSpace& ex, IteratorType first,
-                               IteratorType last, UnaryFunctorType functor) {
+KOKKOS_NVCC_PRAGMA_PUSH
+KOKKOS_NVCC_PRAGMA_SUPPRESS(20011)
+KOKKOS_NVCC_PRAGMA_SUPPRESS(20014)
```

Comment:
@masterleinad what is going on here?

Reply:
It seemed to me that suppressing the warnings is the best way to go (since using `KOKKOS_IF_ON_DEVICE/HOST` leads to more compiler problems). What do you think?